app/model.py

```diff
@@ -5,12 +5,31 @@
 import joblib
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.model_selection import train_test_split
+import json
+
+
+def update_stations_json(df):
+    with open("../utils/stations.json", 'r', encoding="utf-8") as f:
+        stations = json.load(f)
+        # Temporarily store category code to new column
+        df['stationShortCodeCategory'] = df['stationShortCode'].astype("category").cat.codes
+        station_cat = df[['stationShortCode',"stationShortCodeCategory"]].drop_duplicates()
+        for station in stations:
+            category = station_cat[station_cat["stationShortCode"] == station["stationShortCode"]]["stationShortCodeCategory"].values
+            if len(category) == 0:
+                station["stationShortCodeCategory"] = -1
+                continue
+            station["stationShortCodeCategory"] = int(category[0])
+    with open("../utils/stations.json", "w", encoding="utf-8") as f:
+        json.dump(stations, f, ensure_ascii=False)
+
 
 dataset = pd.read_csv('../data/merged/trains_and_weather.csv', low_memory=False)
 
-dataset = dataset.astype({'commuterLineID': 'category', 'stationShortCode': 'category'})
-categoryToNumeric = dataset.select_dtypes(['category']).columns
-dataset[categoryToNumeric] = dataset[categoryToNumeric].apply(lambda x: x.cat.codes)
+update_stations_json(dataset)
+dataset["stationShortCode"] = dataset['stationShortCodeCategory']
+dataset['commuterLineID'] = df['commuterLineID'].astype("category").cat.codes
```

Comment:
Is the df defined here, should it be dataset?

Reply:
a result of bad copypasting it seems